test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -9246,6 +9246,103 @@ def forward(self, x):
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_normal(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Normal(x, y).sample().size(0), x, y
+
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([[1.0], [2.0]])))
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([1.0])))
+
+        self.run_test(M(), (torch.tensor([[[0.0], [10.0]], [[2.0], [8.0]], [[2.0], [8.0]]]), torch.tensor([[1.0], [3.0]])))
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_normal_correctness(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Normal(x, y).sample()
+
+        expected_mean = 5.0
+        expected_var = 10.0
+
+        model_export = M()
+        dummy_input = (torch.tensor([expected_mean]), torch.tensor([expected_var]))
+        ort_sess = convert_to_onnx(model_export, input=dummy_input, opset_version=self.opset_version,
+                                   training=torch.onnx.TrainingMode.EVAL)
+
+        ort_out = []
+        for i in range(1500):
+            ort_out.append(run_ort(ort_sess, input=dummy_input))
+
+        actual_var = np.var(ort_out)
+        actual_mean = np.mean(ort_out)
+
+        assert abs(abs(actual_mean) - expected_mean) <= expected_mean * 0.15, \
+               "the gap of mean between ort outputs and expected one is unacceptable."
+        assert abs(abs(actual_var) - expected_var) <= expected_var * 0.15, \
+               "the gap of variance between ort outputs and expected one is unacceptable."
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_uniform(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Uniform(x, y).sample().size(0), x , y
+
+        self.run_test(M(), (torch.tensor([0.0]), torch.tensor([10.0])))
+        self.run_test(M(), (torch.tensor([[0.0], [6.0]]), torch.tensor([[1.0], [7.0]])))
+        self.run_test(M(), (torch.tensor([1.0]), torch.tensor([[10.0], [7.0], [9.0], [20.0]])))
+
+    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_dist_uniform_correctness(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y):
+                return torch.distributions.Uniform(x, y).sample()
+
+        expected_min = 5.0
+        expected_max = 10.0
+
+        model_export = M()
+        dummy_input = (torch.tensor([expected_min]), torch.tensor([expected_max]))
+        ort_sess = convert_to_onnx(model_export, input=dummy_input, opset_version=self.opset_version,
+                                   training=torch.onnx.TrainingMode.EVAL)
+
+        ort_out = []
+        for i in range(500):
+            ort_out.append(run_ort(ort_sess, input=dummy_input))
+
+        actual_min = min(ort_out)
```

Comment:
Mean of a uniform random variable = 1/2(a+b)
Is it possible to compare the actual mean with expected mean?
Also, expected var =  (b − a)*2/12.

Reply:
No, comparing mean value in this scenario is not right. Uniform() method has a limitation on min and max value, so we should compare them directly if no blocker.

Let's say if we pass min and max value to Uniform() as below:
min: 5
max: 10
Then the mean value should be 7.5((5 + 10)/2).

But if the min and max value are set to below, the mean value could also be 7.5 but the generated values might be out of the above given scope and this should not be acceptable:
min: 6
max: 9